[17.0][MIG] account_invoice_payment_retention: Migration to 17.0

SidroVL opened the pull request.

kittiu committed:
[14.0][ADD] account_invoice_payment_retention

oca-travis committed:
[UPD] Update account_invoice_payment_retention.pot

OCA-git-bot committed:
[ADD] icon.png

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
[UPD] README.rst

Saran440 committed:
[FIX] compute_amount on payment wizard

kittiu committed:
[14.0][FIX] account_invoice_payment_retention, return retention with multi deduction cause error

oca-travis committed:
[UPD] Update account_invoice_payment_retention.pot

OCA-git-bot committed:
account_invoice_payment_retention 14.0.1.0.1

ofonseca-pyming committed:
Added translation using Weblate (Catalan)

OCA-git-bot committed:
[IMP] update dotfiles

ps-tubtim committed:
[14.0][REF] account_invoice_payment_retention

oca-travis committed:
[UPD] Update account_invoice_payment_retention.pot

OCA-git-bot committed:
account_invoice_payment_retention 14.0.1.1.0

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: account-invoicing-14.0/account-invoicing-14.0-account_invoice_payment_retention
Translate-URL: https://translation.odoo-community.org/projects/account-invoicing-14-0/account-invoicing-14-0-account_invoice_payment_retention/

OCA-git-bot committed:
account_invoice_payment_retention 14.0.1.1.1

ps-tubtim committed:
[14.0][ENH] account_invoice_payment_retention: add retention computation method

TheerayutEncoder committed:
[MIG] account_invoice_payment_retention: Migration to 15.0

Saran440 committed:
[ENH] default account retention with customer invoice (receivable account)

Commit:
[UPD] Update account_invoice_payment_retention.pot

OCA-git-bot committed:
[BOT] post-merge updates

Saran440 committed:
[MIG] account_invoice_payment_retention: Migration to 16.0

Commit:
[UPD] Update account_invoice_payment_retention.pot

mymage committed:
Added translation using Weblate (Italian)

Ivorra78 committed:
Added translation using Weblate (Spanish)

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (40 of 40 strings)

Translation: account-invoicing-16.0/account-invoicing-16.0-account_invoice_payment_retention
Translate-URL: https://translation.odoo-community.org/projects/account-invoicing-16-0/account-invoicing-16-0-account_invoice_payment_retention/es/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (40 of 40 strings)

Translation: account-invoicing-16.0/account-invoicing-16.0-account_invoice_payment_retention
Translate-URL: https://translation.odoo-community.org/projects/account-invoicing-16-0/account-invoicing-16-0-account_invoice_payment_retention/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (40 of 40 strings)

Translation: account-invoicing-16.0/account-invoicing-16.0-account_invoice_payment_retention
Translate-URL: https://translation.odoo-community.org/projects/account-invoicing-16-0/account-invoicing-16-0-account_invoice_payment_retention/it/

Commit:
[IMP] account_invoice_payment_retention: pre-commit auto fixes

Commit:
[MIG] account_invoice_payment_retention: Migration to 17.0

SidroVL commented:
In the form view of the wizard (view_account_payment_register_form), in the “apply_payment_retention” field, the widget="boolean_toggle" option has been removed because it causes problems and the value of the field had to be changed several times to have the desired effect.

tv-openbig commented:
@SidroVL Could you please fix this PR so Runboat can build a proper test instance? I’d like to do a functional test. Thanks!

Commit:
[17.0][FIX] account_invoice_payment_retention: fix multi-invoice payment check

The migration commit accidentally changed `active_ids` (plural) to
`active_id` (singular) in action_create_payments(). This made the
multi-invoice retention check unreachable since browse(active_id)
always returns at most one record, so `len(invoices) > 1` was
always False.

Restore the original `active_ids` key so the UserError is properly
raised when paying multiple invoices with retention.

Commit:
[17.0][FIX] account_invoice_payment_retention: use line_ids.move_id in action_create_payments

Replace context-based browse(active_ids) with self.line_ids.move_id
to get the invoices being paid. This is the standard Odoo 17 pattern
already used by _compute_retention_amount_currency and
_validate_payment_retention in the same file.

The previous approach failed when active_ids contained IDs that were
not valid account.move records (e.g. account.move.line IDs from
account_move_cancel_confirm tests), causing a MissingError.

SidroVL commented:
@tv-openbig Done

tv-openbig commented:
@SidroVL Hello , i have reviewed the functionality. It works! Thanks for your contribution and also for your update in order to enable testing.

Usage of retention on final invoice:
 
<img width="1522" height="969" alt="Bildschirmfoto 2026-07-02 um 17 42 30" src="https://github.com/user-attachments/assets/ae1470ab-134a-4c3e-81e5-788350535760" />


Creation of retention:

<img width="1521" height="992" alt="Bildschirmfoto 2026-07-02 um 17 42 46" src="https://github.com/user-attachments/assets/d9cac30d-fadc-4c59-a5d1-f590b89bd2f8" />
